.github/workflows/check.yml

```diff
@@ -91,6 +91,57 @@ jobs:
         run: |
           npm run test
 
+  stage:
+    needs: queue
+    runs-on: ubuntu-latest
+    env:
+      CLOUD_BUILD_KEY: ${{ secrets.CLOUD_BUILD_KEY }}
+      CLOUD_BUILD_SECRET: ${{ secrets.CLOUD_BUILD_SECRET }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+    steps:
+      - name: Posting announcement
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          message: |
+            Staging build in progress ...
+      - name: Getting changes
+        id: changes
+        uses: dorny/paths-filter@v2
+        with:
+          list-files: json
+          filters: |
+            app:
+              - 'package.json'
+              - 'server/**/*.js'
+            static:
+              - 'site/**/*'
+
+      - name: Cloning repository
+        uses: actions/checkout@v3
+
+      - name: Setting up Node.js
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version: 16
+
+      - name: Installing Node.js packages
+        run: npm ci
+
+      - name: Staging
+        id: staging
+        env:
+          CHANGED_APP_FILES: ${{ steps.changes.outputs.app_files }}
```

Comment:
Yep indeed!

> If enabled, for each filter it sets an output variable with the name ${FILTER_NAME}_files. It will contain a list of all files matching the filter

from https://github.com/dorny/paths-filter#outputs.

Reply:
Not using dorny/paths-filter anymore as this now requires a file-changed method that works both on Cloud Build and GitHub actions, so I am simply querying the GitHub API and using gulp tasks to filter.